src/Datagrid/ListMapper.php

```diff
@@ -92,6 +92,17 @@ public function add($name, $type = null, array $fieldDescriptionOptions = [])
             $fieldDescriptionOptions['virtual_field'] = true;
         }
 
+        if (\array_key_exists('identifier', $fieldDescriptionOptions) && !\is_bool($fieldDescriptionOptions['identifier'])) {
+            @trigger_error(
+                'Passing a non boolean value for "identifier" option is deprecated since 3.x and will throw an exception in 4.0.',
```

Comment:
l
```suggestion
                'Passing a non boolean value for the "identifier" option is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
```